Export Analysis Reports/Sdl.Community.ExportAnalysisReports/Service/ReportService.cs

```diff
@@ -0,0 +1,355 @@
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Newtonsoft.Json;
+using Sdl.Community.ExportAnalysisReports.Helpers;
+using Sdl.Community.ExportAnalysisReports.Interfaces;
+using Sdl.Community.ExportAnalysisReports.Model;
+using Sdl.ProjectAutomation.Core;
+
+namespace Sdl.Community.ExportAnalysisReports.Service
+{
+	public class ReportService : IReportService
+	{
+		public static readonly Log Log = Log.Instance;
+		private readonly string _communityFolderPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "SDL Community", "ExportAnalysisReports");
+		private readonly Help _help;
+		private readonly IMessageBoxService _messageBoxService;
+		private readonly IStudioService _studioService;
+
+		public ReportService(IMessageBoxService messageBoxService, IStudioService studioService)
+		{
+			_help = new Help();
+			_messageBoxService = messageBoxService;
+			_studioService = studioService;
+		}
+
+		public string JsonPath => Path.Combine(_communityFolderPath, "ExportAnalysisReportSettings.json");
+		public string ReportsFolderPath { get; set; }
+		/// <summary>
+		/// Check if the report was successfully generated
+		/// </summary>
+		/// <param name="reportOutputPath"></param>
+		/// <param name="isChecked"></param>
+		/// <param name="optionalInformation"></param>
+		/// <param name="projects"></param>
+		/// <returns></returns>
+		public bool GenerateReportFile(BindingList<ProjectDetails> projects, OptionalInformation optionalInformation, string reportOutputPath, bool isChecked)
+		{
+			try
+			{
+				_help.CreateDirectory(reportOutputPath);
+				var projectsToBeExported = projects.Where(p => p.ShouldBeExported).ToList();
+				var areCheckedLanguages = projectsToBeExported.Any(p => p.PojectLanguages.Any(l => l.Value));
+				if (!areCheckedLanguages && projectsToBeExported.Count >= 1)
+				{
+					_messageBoxService.ShowInformationMessage(PluginResources.SelectLanguage_Export_Message, PluginResources.ExportResult_Label);
+					return false;
+				}
+				foreach (var project in projectsToBeExported)
+				{
+					// check which languages to export
+					if (project.PojectLanguages != null)
+					{
+						var checkedLanguages = project.PojectLanguages.Where(l => l.Value).ToList();
+
+						foreach (var languageReport in checkedLanguages)
+						{
+							if (string.IsNullOrEmpty(project.ReportPath))
+							{
+								project.ReportPath = reportOutputPath;
+							}
+
+							WriteReportFile(project, optionalInformation, languageReport, isChecked);
+						}
+					}
+				}
+				return true;
+			}
+			catch (Exception exception)
+			{
+				Log.Logger.Error($"GenerateReport method: {exception.Message}\n {exception.StackTrace}");
+				throw;
+			}
+		}
+
+		/// <summary>
+		/// Get report output path from Json file
+		/// </summary>
+		/// <param name="jsonPath"></param>
+		/// <returns></returns>
+		public string GetJsonReportPath(string jsonPath)
```

Comment:
Can be shortened by using "??" and moving the declaration of item in outer scope.

Reply:
The method was refactored and it can be check within the latest commit